### PR TITLE
37051 : fix redirect URL after clicking on the space logo

### PR DESCRIPTION
**Pull request by GouadriaHanen**
after changing the space name the URL was created ":spaces:space_new_name/space_new_name"
fixed to  ":spaces:space_group_Id/space_new_name"

### CODE REVIEW

File: application/src/main/webapp/vue-app/components/ExoChatContact.vue [hunk 1]
```diff
@@ -183,10 +193,10 @@ export default {
       this.isOnline = false;
     },
     getProfileLink() {
-      if (this.app && this.type === 'u') {
```
Comment (ahamdi) [inline application/src/main/webapp/vue-app/components/ExoChatContact.vue:186]: why did you remove the check on this.app ?

File: application/src/main/webapp/vue-app/components/ExoChatContact.vue [hunk 2]
```diff
@@ -117,7 +117,8 @@ export default {
         donotdisturb: this.$t('exoplatform.chat.donotdisturb'),
         invisible: this.$t('exoplatform.chat.invisible'),
         offline: this.$t('exoplatform.chat.button.offline')
-      }
+      },
+      spaceGroupUri: null,
```
Comment (ahamdi) [inline application/src/main/webapp/vue-app/components/ExoChatContact.vue]: No need to add this as data

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: application/src/main/webapp/vue-app/components/ExoChatContact.vue (hunk 2 above)
Comment (GouadriaHanen) [reply application/src/main/webapp/vue-app/components/ExoChatContact.vue]: it is a global variable when it is putted into the method its value never changed

File: application/src/main/webapp/vue-app/components/ExoChatContact.vue (hunk 1 above)
Comment (GouadriaHanen) [reply application/src/main/webapp/vue-app/components/ExoChatContact.vue:186]: app is undefined

File: application/src/main/webapp/vue-app/components/ExoChatContact.vue [hunk 3]
```diff
@@ -173,6 +174,15 @@ export default {
...
   },
   methods: {
+    getSpaceURI() {
+      const spaceId = this.name.toLowerCase().split(' ').join('_');
+      getSpaceByPrettyName(this.name).then((space) => {
+        if (space && space.identity) {
+          this.spaceGroupUri= space.groupId.replace(/\//g, ':');
+        }
+      });
+      return `${eXo.env.portal.context}/g/${this.spaceGroupUri}/${spaceId}`;
```
Comment (aelbenney) [inline application/src/main/webapp/vue-app/components/ExoChatContact.vue]:
```suggestion
      return getSpaceByPrettyName(this.name).then((space) => {
        if (space && space.identity) {
          this.spaceGroupUri= space.groupId.replace(/\//g, ':');
          return `${eXo.env.portal.context}/g/${this.spaceGroupUri}/${spaceId}`;
        }
      });
```

Comment (aelbenney) [reply application/src/main/webapp/vue-app/components/ExoChatContact.vue]: it will return the value before the promise is fulfilled and change `spaceGroupUri` value.

**Review by aelbenney — CHANGES_REQUESTED** [review]

**Review by hbenali — COMMENTED** [review]
### [Recommendation]
 You better short the PR's title. Thanks

**Review by ahamdi — DISMISSED** [review]

File: application/src/main/webapp/vue-app/chatServices.js [hunk 4]
```diff
@@ -413,6 +413,24 @@ export function getSpaceProfileLink(space) {
   // FIXME very ugly, the technical ID should be used here instead
   const spaceId = space.toLowerCase().split(' ').join('_');
   return `${chatConstants.PORTAL}${chatConstants.PROFILE_SPACE_LINK}${spaceId}/${spaceId}`;
+
```
Comment (aelbenney) [inline application/src/main/webapp/vue-app/chatServices.js]: Add this empty line after this method.

**Review by aelbenney — CHANGES_REQUESTED** [review]

**Review by aelbenney — APPROVED** [review]
